### ISSUE TITLE
Working example code to avoid PrefixAllGlobals and I18n

### ISSUE THREAD

**Issue by mikaelz** [WordPress-Coding-Standards/WordPress-Coding-Standards#1109]
It would be great to have a wiki entry with examples how to avoid [PrefixAllGlobals](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/aebfb7a479f6f249cf6aa0c3d5bc8605768cb89a/phpcs.xml.dist.sample#L71) and [I18n](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/aebfb7a479f6f249cf6aa0c3d5bc8605768cb89a/phpcs.xml.dist.sample#L65) rule warnings.

I couldn't fix the warnings in my projects so I commented the rules :/

Thanks in advance.

**Comment by jrfnl** [comment]
Could you give some example code where you were having trouble fixing the issues with it ? That would be helpful as a base to create the articles.

**Comment by mikaelz** [comment]
Oh, sure.
phpcs complains with **Functions declared by a theme/plugin should start with the theme/plugin prefix. Found: "mz_woocommerce_stock_catalog".**

For
```
function mz_woocommerce_stock_catalog( $variation_id = '', $qty = null ) {
```
Rule in my phpcs.xml
```
<rule ref="WordPress.NamingConventions.PrefixAllGlobals">
  <properties>
	<property name="prefixes" type="array" value="my_prefix"/>
  </properties>
</rule>
```

Now I see **value="my_prefix"**. Changing mz_woocommerce_stock_catalog to my_prefix_woocommerce_stock_catalog solves the issue.

For `__( 'Nie je na sklade', 'mz' );` I get the **Mismatch text domain. Expected 'my-textdomain' or 'library-textdomain' but got 'mz'.** message

In phpcs.xml
```
<rule ref="WordPress.WP.I18n">
  <properties>
	<property name="text_domain" type="array" value="my-textdomain,library-textdomain"/>
  </properties>
</rule>
```

Sorry, my bad. But I think it would be great to have examples.

**Comment by GaryJones** [comment]
Here's the relevant code you'll likely want in your theme's `phpcs.xml.dist`:

```xml
<rule ref="WordPress.NamingConventions.PrefixAllGlobals">
  <properties>
	<property name="prefixes" type="array" value="mz"/>
  </properties>
</rule>
<rule ref="WordPress.WP.I18n">
  <properties>
	<property name="text_domain" type="array" value="mz"/>
  </properties>
</rule>
```

In this case, all I've done is changed the `value`s in each property, to match up with your code examples - a prefix of `mz` for functions, and a textdomain of `mz` (yes, the same value) for i18hn functions.

Obviously, these are custom properties for your theme, so the example / docs can only make suggestions (for those developers who are comfortable with English, the use of "my" often indicates where some value needs to be customised), with the expectation that folks understand to adapt the exact value to their needs.

There are other bits of `phpcs.xml.dist.sample` you'll probably want to update as well, so check it over carefully, and fix as needed.
